### PR TITLE
New Tracks UI

### LINKED ISSUE #503: New Tracks UI
Screenshot 2023-08-29 at 9 59 28 AM

### PR DESCRIPTION
**Pull request by SubxX**
## Description
Revamped old tracks design to a new design similar to leet code study plan section

## Related Issue
closes #503 

## Screenshots/Video (if applicable):
Old Design

image

New Design

Explore Tracks _ TypeHero (1)

### CODE REVIEW

**Comment by bautistaaa** [comment]
looks good so far!

we still need an empty state or something to let the user know they have not enrolled in any tracks yet. 

to add on to that i think there should be an easy way to enroll from the tracks page so they don't have to click in and if they are enrolled in a course some indicator on the card itself (enrolled badge or something)

**Comment by bautistaaa** [comment]
pulse to check to see if this PR is still alive

**Comment by SubxX** [comment]
Yes its alive @bautistaaa 
Will complete this by today the currently enrolled badge is implemented. Only the empty state is remaining.

**Comment by SubxX** [comment]
@bautistaaa 

**Here is the UI for empty data**

image

**And here is the enrolled badge.**

image


Please let me know your thoughts
and about your last feedback 

>  i think there should be an easy way to enroll from the tracks page so they don't have to click in

Right now our tracks card is so small if we try to fit in a button it looks very bad and out or proportion, I am still thinking for a solution if you have something in mind please share with me

**Comment by bautistaaa** [comment]
i think this is a good base to iterate on.

i still think the empty state is too big but this is way better than whats there

File: apps/web/src/app/[locale]/tracks/_components/track-card.tsx [hunk 1]
```diff
@@ -1,56 +1,86 @@
...
+          isEnrolled
+            ? 'border-blue-400'
+            : 'group-hover:border-blue-400 group-focus:border-blue-400',
+        )}
+      >
+        {track.enrolledUsers?.length ? <EnrolledBadge /> : null}
+        <CardContent className="flex items-center gap-5 p-2 pr-4">
+          <div
+            className={clsx(
+              `bg-gradient-to-r from-neutral-500/10 from-10% ${BGS_BY_TRACK[index]} relative to-100% dark:from-neutral-500/20`,
```
Comment (bautistaaa) [inline apps/web/src/app/[locale]/tracks/_components/track-card.tsx:56]: does index actually correlate with difficulty here?

**Review by bautistaaa — APPROVED** [review]

File: apps/web/src/app/[locale]/tracks/_components/track-card.tsx (hunk 1 above)
Comment (SubxX) [reply apps/web/src/app/[locale]/tracks/_components/track-card.tsx:56]: No there is no difficulty level set for a track. Here the index denotes the iteration count of the `tracks.map`.

We will have max 10 tracks and I don't know what will be the name/slug of the track so for now I used the element index values for assigning the unique gradient.
@bautistaaa

**Pull request merged**